### PR TITLE
RavenDB-22874 - Tombstones from artificial documents are being replic…

### PR DESCRIPTION
**Pull request by shiranshalom**
…ated

### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-22874/Tombstones-from-artificial-documents-are-being-replicated

### Type of change

- [x] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [x] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [ ] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

File: src/Raven.Server/Documents/DocumentsStorage.cs [hunk 1]
```diff
@@ -1709,6 +1709,10 @@ public static Tombstone TableValueToTombstone(JsonOperationContext context, ref
                 collectionName = ExtractCollectionName(context, doc.Data);
 
                 var flags = GetFlagsFromOldDocument(newFlags, doc.Flags, nonPersistentFlags);
+
+                if (doc.Flags.Contain(DocumentFlags.Artificial))
```
Comment (ppekrol) [inline src/Raven.Server/Documents/DocumentsStorage.cs]: How about renaming `GetFlagsFromOldDocument` to:
1. GetFlagsFromOldDocumentForPut
2. GetFlagsFromOldDocumentForDelete

With impl as follows:
```

GetFlagsFromOldDocumentForDelete()
{
    var flags = GetFlagsFromOldDocumentForPut();
    if (doc.Flags.Contain(DocumentFlags.Artificial))}
         flags |= DocumentFlags.Artificial;

    return flags;
}
```

Comment (shiranshalom) [reply src/Raven.Server/Documents/DocumentsStorage.cs]: done

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Server/Documents/DocumentsStorage.cs [hunk 2]
```diff
@@ -2689,6 +2690,15 @@ public DocumentFlags GetFlagsFromOldDocument(DocumentFlags newFlags, DocumentFla
             return newFlags;
         }
 
+        public DocumentFlags GetFlagsFromOldDocumentForDelete(DocumentFlags newFlags, DocumentFlags oldFlags, NonPersistentDocumentFlags nonPersistentFlags)
+        {
+            var flags = GetFlagsFromOldDocumentForPut(newFlags, oldFlags, nonPersistentFlags);
+            if (oldFlags.Contain(DocumentFlags.Artificial))
+                flags |= DocumentFlags.Artificial;
```
Comment (ppekrol) [inline src/Raven.Server/Documents/DocumentsStorage.cs:2697]: Assuming that we do not need `FromIndex` flag?

Comment (shiranshalom) [reply src/Raven.Server/Documents/DocumentsStorage.cs:2697]: yep, I don't think we do...

**Pull request merged**